### DISCUSSION TITLE
Getting Started

### DISCUSSION THREAD

**Discussion by waseem-medhat** [PracticeCraft/roadmap#3] [General]
The [org's roadmap](https://github.com/PracticeCraft/roadmap) got me interested to be part of this project. I am equally interested in helping and getting some help myself. So, I'd like to put in here some ideas on how I can do so, and I'll look forward to any thoughts shared in the discussion.

So, here's what I have in mind...

1. **Get the conversation going.** This is a collaborative project, so it should be clear that community involvement is of critical importance. I can reach out to people within my network, maybe posts/discussions on Discord, stuff like that.
2. **Think about a general plan + project ideas**. I'm simply ready to propose ideas and brainstorm with whoever is part of the collaboration.
3. **Write code!** As of now, my focus is on the backend and Go in particular. So, whatever projects we have in mind, I can start working on their backends as soon as the idea is fleshed out. But I might still be able to provide a little support for frontend devs.

Looking forward to your thoughts!

**Comment by nikfp** [comment]
@wipdev-tech glad you found this interesting! 

I'd love to see this get off the ground and I've had a lot of ideas since I wrote that roadmap. However I've had very little time or energy to put into anything tech related for quite a while, and I'm not sure when and to what degree that's going to change. I think your first point is right on the money though, the conversation needs to start. So let's start it. 

I see the following things required to get this rolling and to the point that it can sustain itself. 

1. **I need help with the project in it's infancy, simply because of my lack of time to apply to it**. If you are willing to take the time and engage with this, I'm even fine with adding you as an admin on the GH org so you can start making things move as you see fit. 
2. A few things need to be first priorities in my mind. Number one would be a code of conduct that is inclusive and fair, followed by some basic work for things like a landing page and some resources to use when advocating for the project. 
3. Start sharing this in programming circles and see if it resonates. In particular the project will need to target people with _some_ team based experience in the beginning, so we can mine their knowledge of how to get this thing into a shape that will either just work out, or will at least be flexible enough to adjust as needed. 

As stated, I don't know how much my particular involvement can be at the moment. My strong preference would be A LOT but that won't be the reality for a while. So personally my first goal would be assembling the initial "core team" that can take this from just another idea to something that helps people. I don't want to over-commit and have that be the reason this fails, which is why it's sat for almost a year, and I don't want anyone else doing that either. So having some depth in a team would be a huge benefit. 

Let me know if you want to jump in and start working, noting what I've just said, and I'll get you added if you do. I'm willing to give free reign to the first handful of people getting involved so they can make some decisions and advance this without me being a blocker.

**Reply by waseem-medhat** [reply]
Thanks for sharing your thoughts @nikfp! I think I'll have some time in the next couple of months to be able to contribute. So, you can add me to the org, and I'll do what I can in that time to kick things off.

As for the specific points you mentioned (CoC, landing page, resources). I can start writing some drafts right now... I'm thinking it could be another repo like `github.com/PracticeCraft/getting-started` with both the CoC and relevant resources for new contributors. But in that case, do we really need a landing page? (unless it's a very simple one and points to the documentation repo(s))

As for your involvement, the more the better obviously, but with limited time/energy I won't need more than your feedback or some brainstorming like we're doing here - at your convenience.

How about that?

**Reply by nikfp** [reply]
This sounds like it will work well. I also agree with getting the "onboarding" materials in their own repo. Separation of concerns and so on. 

The landing page could be very simple in my opinion, even as simple as a GH page that allows people to share a link to a curated entry point. I can't see requiring anything beyond a static HTML / CSS page for now, but later on that might turn into a gateway for .... something? I don't want to get ahead of myself. 

I'll engage as much as I can, when I can. Asynchronous communication will probably work best like we have going now. I'll also get you added to the org. 

Thanks a million for reaching out! I hope we can make something out of this.

**Reply by nikfp** [reply]
@wipdev-tech you are in as a member and I gave you high level access. With great power comes great responsibility. 

I'd also like to hear your take on what would be good to tackle first. And I also noticed the roadmap repo has accumulated some stars, that might be a source of interested parties.

**Reply by waseem-medhat** [reply]
IIRC GitHub has a way of making a Kanban board, and there we can lay down all our thoughts and move them around in that Kanban style (e.g., planned, in progress, done). I think that when all the goals are visible (and public) we can assign them to whoever has the time (me, you, someone else who joins later), see how things work and iterate/improve as we go.

That's the closest I got to a tangible plan :joy:

**Reply by waseem-medhat** [reply]
@nikfp I've started putting some stuff together in the Projects section, but right now it's still private to the org until you take a look and let me know if it looks good enough for us to go forward with it that way. Until then, I'll be filling the cards with my ideas.

P.S. thanks for adding me here!

**Comment by nikfp** [comment]
@wipdev-tech I got a chance to look at that and it looks like exactly what we need. I'll consider what I might have some time to engage with and circle back.
